middleware/src/middleware.go

```diff
@@ -181,8 +182,26 @@ func (middleware *Middleware) updateCheckLoop() {
 	}
 }
 
+// hsmHeartbeatLoop
+func (middleware *Middleware) hsmHeartbeatLoop() {
+	for {
+		// TODO(@0xB10C) fetch the `stateCode` and `descriptionCode` from redis keys set byt the supervisor
+		err := middleware.hsmFirmware.BitBoxBaseHeartbeat(messages.BitBoxBaseHeartbeatRequest_IDLE, messages.BitBoxBaseHeartbeatRequest_EMPTY)
+		if err != nil {
+			log.Printf("Received an error from the HSM: %s\n", err)
+			continue
```

Comment:
Do we really want to continue here without waiting? If the next heartbeats fail as well we end up flooding the HSM with heartbeats.

Reply:
Good call, I can add a 1-second sleep there before the continue.

Reply:
Yeah, but should coordinate if we want to merge this PR or if we want to have it all in one.

Reply:
no strong preference from my side.

Reply:
Updated and rebased.
Fine with me either way as well.
I think in that case, let's just merge it.